### PR TITLE
Date input commit on change

### PR DESCRIPTION
**Pull request by lbwexler**
Hoist P/R Checklist
-------------------

**Pull request authors:** Review and check off the below. Items that do not apply can also be
checked off to indicate they have been considered. If unclear if a step is relevant, please leave
unchecked and note in comments.

- [ ] Caught up with `develop` branch as of last change.
- [ ] Added CHANGELOG entry, or determined not required.
- [ ] Reviewed for breaking changes, added `breaking-change` label + CHANGELOG if so.
- [ ] Updated doc comments / prop-types, or determined not required.
- [ ] Reviewed and tested on Mobile, or determined not required.
- [ ] Created Toolbox branch / PR, or determined not required.

**If your change is still a WIP**, please use the "Create draft pull request" option in the split
button below to indicate it is not ready yet for a final review.

> **Pull request reviewers:** when merging this P/R, please consider using a *squash commit* to
> collapse multiple intermediate commits into a single commit representing the overall feature
> change. This helps keep the commit log clean and easy to scan across releases. PRs containing a
> single commit should be *rebased* when possible.

### CODE REVIEW

File: desktop/cmp/input/DateInput.js [hunk 1]
```diff
@@ -239,9 +273,8 @@ class Model extends HoistInputModel {
     };
 
     onKeyDown = (ev) => {
-        if (ev.key === 'Enter') {
+        if (ev.key === 'Enter' || ev.key === 'Tab') {
             this.doCommit();
-            this.consumeEvent(ev);
```
Comment (amcclain) [inline desktop/cmp/input/DateInput.js:244]: What issues could we see from dropping this?

File: desktop/cmp/input/DateInput.js [hunk 2]
```diff
@@ -43,6 +43,9 @@ DateInput.propTypes = {
     ...HoistInputPropTypes,
     value: PT.oneOfType([PT.instanceOf(Date), PT.instanceOf(LocalDate)]),
 
+    /** True to commit on every change/keystroke, default false. */
```
Comment (amcclain) [inline desktop/cmp/input/DateInput.js:46]: I think this doc string will need more elaboration re. any special behaviors in the context of the date input, where the text input might not be valid, etc.

**Review by amcclain — COMMENTED** [review]
Hi John - thanks for looking at this.

I see you’ve added `commitOnChange` with the standard default of false, but then it doesn’t appear that we’re setting to true in `DateEditor` - what’s the story there?  Was that not really needed? 

I am a bit thrown off by the date picker appearing but without the picker icon.  I suppose it’s not a big deal - are we simply unable to get the picker icon to align OK in the grid context, or are there other issues?

(Note this also breaks support for `enableClear` prop - maybe not that important.)

I wish I were feeling better about this set of components - they have given us issues in the past, and I still don't personally feel that I have a good grasp or understanding of how they work and what the options we expose are doing.

File: desktop/cmp/grid/editors/impl/InlineEditorModel.js [hunk 3]
```diff
@@ -47,7 +46,7 @@ export function useInlineEditorModel(component, props, ref, isPopup = false) {
         model: impl,
         bind: 'value',
         commitOnChange: true,
-        ref: composeRefs(ref, impl.ref),
+        ref: impl.ref,
```
Comment (lbwexler) [inline desktop/cmp/grid/editors/impl/InlineEditorModel.js:49]: this seems correct to me -- surprised it was not causing us even more problems before

**Comment by lbwexler** [comment]
The `commitOnChange:true` setting happens on all the editors in the hook

**Comment by lbwexler** [comment]
closing in favor of the alternative PR we have been working on

**Pull request closed without merging**